### PR TITLE
Store additional config in the riak.conf

### PR DESCRIPTION
**Pull request by bostko**
- additional config moved from the RiakEnterpriseNode
- bump up default Riak version

### CODE REVIEW

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java [hunk 1]
```diff
@@ -251,6 +251,15 @@ public void customize() {
             String saveAsRiakConf = Urls.mergePaths(getRunDir(), "riak.conf");
             DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsRiakConf).contents(riakConfTemplate));
             commands.add(sudo("mv " + saveAsRiakConf + " " + getRiakEtcDir()));
+
+            if(!osDetails.isMac()) {
```
Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: Why leave out additional config for Mac?

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java [hunk 2]
```diff
@@ -251,6 +251,15 @@ public void customize() {
...
+                if(!com.google.common.base.Strings.isNullOrEmpty(entity.getConfig(RiakNode.RIAK_CONFIG_FILE))) {
```
Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: Isn't it better to leave the default to be empty string and check for non-null only here? Any reason empty string should be treated specially?

Comment (bostko) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: There is no point in doing this task if the string is empty

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java [hunk 3]
```diff
@@ -251,6 +251,15 @@ public void customize() {
...
+                    String configTemplate = processTemplateContents(entity.getConfig(RiakNode.RIAK_CONFIG_FILE));
+                    String saveAsConfig = Urls.mergePaths(getRunDir(), "riak_extra.conf");
+                    DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsConfig).contents(configTemplate));
+                    commands.add(sudo("cat " + saveAsConfig + " >> " + getRiakEtcDir() + "/riak.conf"));
```
Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: `customize` step is called on each start, which will lead to multiple copies of the config appended.

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java (hunk 1 above)
Comment (bostko) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: I see that there is RIAK_CONF_TEMPLATE_URL_MAC and RIAK_CONF_TEMPLATE_URL_LINUX .
I am not familiar with differences between Riak conf on Mac and Linux but I guess your are right and we can use RIAK_CONFIG_FILE for both.

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java [hunk 4]
```diff
@@ -251,6 +251,15 @@ public void customize() {
             String saveAsRiakConf = Urls.mergePaths(getRunDir(), "riak.conf");
             DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsRiakConf).contents(riakConfTemplate));
             commands.add(sudo("mv " + saveAsRiakConf + " " + getRiakEtcDir()));
+
+            if(!osDetails.isMac()) {
+                if(!com.google.common.base.Strings.isNullOrEmpty(entity.getConfig(RiakNode.RIAK_CONFIG_FILE))) {
+                    String configTemplate = processTemplateContents(entity.getConfig(RiakNode.RIAK_CONFIG_FILE));
+                    String saveAsConfig = Urls.mergePaths(getRunDir(), "riak_extra.conf");
+                    DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsConfig).contents(configTemplate));
```
Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: Driver methods are executed in effector context so it's safe to use DynamicTasks.queue (somebody correct me if I am wrong?). At the very least you should follow `queueIfPossible` with `orSubmitAsync` to be sure the task is executed.

File: software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java (hunk 3 above)
Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: Oh I see the config is uploaded above, so should be safe to cat it here.

Comment (neykov) [inline software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java]: But note that if `isVersion1()` then the config file is called `app.config`. I think it's better to prepare the config once and then push it to the machine, will save you from needing to execute bash commands to append.

**Comment by neykov** [comment]
Never mind the previous two comments, I was looking at a single commit only. Looks good now. Squash the commits so the history is cleaner, then can be merged.

**Comment by ahgittin** [comment]
LGTM

**Pull request merged**